it/basic_test.py

```diff
@@ -33,3 +33,11 @@ def test_run_with_help(cfg):
     output = process.run_subprocess_with_output(cmd)
     expected = "usage: esrally [-h] [--version]"
     assert expected in "\n".join(output)
+
+
+@it.rally_in_mem
+def test_run_without_http_connection(cfg):
+    cmd = it.esrally_command_line_for(cfg, "list races")
+    output = process.run_subprocess_with_output(cmd, {"http_proxy": "http://non-existent-proxy-server.com"})
```

Comment:
Can you please use `http://invalid` instead of `http://non-existent-proxy-server.com`? It's reserved so no one will be able to register it: https://en.wikipedia.org/wiki/.invalid (Unlikely, but this happened to me in the past: I was using a GCS bucket name that did not exist in tests, eventually someone used it and it broke our CI.)

Reply:
Changed.